core/Cargo.toml

```diff
@@ -19,11 +19,15 @@ include = ["/Cargo.toml", "src/**/*.rs", "/README.md", "/LICENSE"]
 
 [dependencies]
 pdsl_utils = { path = "../utils/" }
-parity-codec = { version = "3.2", default-features = false, features = ["derive", "full"] }
+parity-codec = { version = "3.3", default-features = false, features = ["derive", "full"] }
 log = "0.4"
 wee_alloc = { version = "0.4", default-features = false }
+node-runtime = { git = "https://github.com/paritytech/substrate/", branch = "v1.0", package = "node-runtime", default-features = false }
```

Comment:
So having `node-runtime` in our dependency tree means that we now have a shared dependency base with it which is huge I assume. Since `cargo` cannot properly unshare crate features of dependencies we have to be really careful about overlapping ones.

Reply:
We can remove `wee_alloc` dependency here.
Also `pdsl_core` should not depend on `syn`.